unblob/processing.py

```diff
@@ -737,8 +737,7 @@ def calculate_block_size(
     # We don't care about floating point precision here
     block_size = file_size // chunk_count
     block_size = max(min_limit, block_size)
-    block_size = min(block_size, max_limit)
-    return block_size
+    return min(block_size, max_limit)
```

Comment:
```
    - unblob/processing.py:
        1 × RET504 (unnecessary-assign)
```
https://docs.astral.sh/ruff/rules/unnecessary-assign/

Reply:
I understand what is written there, but feel it is a rather weak-to-nonexistent argument, and the end result of the automatic fixer

```
def calculate_block_size(
...
    return min(block_size, max_limit)
```

simply feels wrong.

The variable `block_size` is calculated in multiple steps with each improving on it by enforcing a constraint, and I would expect the final version of it to be returned holding the best version of the multi-step refinement, but this is not what happens, as a last refinement happens in the `return`. So going with this "fix", would require renaming `block_size` to something else as well (`almost_block_size`?).

This ruff rule feels to be a special case for inlining (an optimization), and there could be other arbitrary rules for inlining.
E.g. in this case the `block_size` variable could fully inlined as the previous lines are just simple transformations of the previous value.
If we do it, we get

```py
    return min(max(min_limit, file_size // chunk_count), max_limit)
```

It is equivalent to the original, it was a simple transformation, it is not even that complex. It is definitely much shorter than the original. Some people would write it as such, some even would not dedicate a function for it. However I still feel it much less understandable than the original, because we lose the sense of understanding its step-wise improvement logic, which makes it much harder to verify/understand/maintain.

----

I do not feel enforcing this rule adds to code quality.
I expect these kinds of optimizations should be done at the compiler level, not enforced on humans, because it might hurt expression of thought.
Could you please revert this change and disable this check?

Reply:
This check was [there](https://github.com/onekey-sec/unblob/commit/33e3d6ddf628cf1906c18f774d475813c717138d) when we originally implemented ruff and rewritten concidreable amount of code due to it. It is now just catches this case as well.
I tend to disagree with you on this (although still don't have a strongly held oppinion TBH). I know I am in the `calculate_block_size` function, so it does not add any value that the return value is called `block_size`. Although, having the partial computation result called `block_size` is feels more wrong.

Reply:
Note that I don't argument citing a priory decision here, but the (lack of?) value needs to be viewed in context, and I yet to see a case where having the return value assigned to a variable improved readability instead of producing clutter. (for me)